docs/plugin-eval.md

```diff
@@ -0,0 +1,221 @@
+  ],
+  "input": [
+    {
+      "role": "system",
+      "content": "Answer the following questions as concisely as possible."
+    },
+    {
+      "role": "user",
+      "content": "Can you list all of the birds in my birds list which are in the Accipitridae family?"
+    },
+    {
+      "role": "plugin",
+      "namespace": "MyBirds",
+      "endpoint": "listBirds",
+      "content": {}
+    }
+  ]
+}
+```
+
+## Plugin Creation
+
+### Namespace and API Description
+
+- Use the `@namespace` decorator to define your plugin's namespace and provide a brief description. The AI model will use this namespace to encapsulate your plugin. The description will help the model understand the goals, purposes, and limitations of your plugin.
+
+- For each API method in your plugin, use the `@api_description` decorator to define the API's name, description, and arguments (if any). The AI model will use this information to understand the purpose of each API method and the inputs it expects. The description will also provide context for the function.
+
+### Input and Output
+
+- Ensure your plugin methods accept inputs as defined in the `api_args` parameter of the `@api_description` decorator.
+
+- The output of your plugin methods should be in a format that can be easily consumed by the AI model.
+
+### Adding to Registry
+
+- To add a plugin to the registry, create a new YAML file in the `registry/plugins` directory under the same directory structure and name as your plugin Python file.
+
+- The naming convention for a new registry entity is `<namespace>.<plugin_name>.<version>`. Feel free to be creative with the namespace and plugin name. The version is a monotonically increasing integer, starting at 1. For example, `openai.bird.1`.
+
+- The YAML file should contain a description of how to instantiate your plugin. A few examples can be found at `registry/plugins/openai/example.yaml`.
+
+### Evaluation Examples
+
+- Create evaluation examples that involve your custom plugin. Plugins should be enabled on a per-example basis, allowing us to group examples of how the plugin changes the model's behavior with examples that might not use the plugin at all but are related requests.
+
+- Model-graded evaluations tend to work better than string-based evaluations.
+
+- Make sure your evaluation passes the `plugins` dictionary to the sampling command. This will ensure that the model is aware of your plugin and can use it during local evaluation.
+
+- It is best-practice to test the behavior of the model when your plugin is not installed. This helps effectively communicate the value of your plugin when combined with the model.
+
+- We've found evaluating the appropriateness of plugin use, the model's ability to recover from inappropriate or adversarial responses, and the ability for the model to integrate the plugin's knowledge into its own understanding of the world to be important evaluation dimensions.  You may find other dimensions to be important as well.
```

Comment:
Would be great to have examples for each of these use cases and how to implement that